Branch-SDK/BNCSKAdNetwork.m

```diff
@@ -91,4 +101,88 @@ - (void)updatePostbackConversionValue:(NSInteger)conversionValue
     
 }
 
+- (void)updatePostbackConversionValue:(NSInteger)fineValue
+                          coarseValue:(NSString *)coarseValue
+                           lockWindow:(BOOL)lockWindow
+                    completionHandler:(void (^)(NSError *error))completion {
+    if (@available(iOS 16.1, *)) {
+        if ([self shouldAttemptSKAdNetworkCallout]) {
+            
+            ((id (*)(id, SEL, NSInteger, NSString *, BOOL, void (^)(NSError *error)))[self.skAdNetworkClass methodForSelector:self.skAdNetworkUpdatePostbackConversionValueCoarseValueLockWindow])(self.skAdNetworkClass, self.skAdNetworkUpdatePostbackConversionValueCoarseValueLockWindow, fineValue, coarseValue, lockWindow, completion);
+        }
+    }
+}
+
+- (int) calculateSKANWindowForTime:(NSDate *) currentTime{
+    
+    int firstWindowDuration = 2 * 24 * 3600;
+    int secondWindowDuration = 7 * 24 * 3600;
+    int thirdWindowDuration = 35 * 24 * 3600;
+    
+    NSTimeInterval timeDiff = [currentTime timeIntervalSinceDate:[BNCPreferenceHelper sharedInstance].firstAppLaunchTime];
+    
+    if (timeDiff <= firstWindowDuration) {
+        return 1;
+    } else if (timeDiff <= secondWindowDuration) {
+        return 2;
+    }else if (timeDiff <= thirdWindowDuration) {
+        return 3;
+    }
+    return 0;
+}
+
+- (NSString *) getCoarseConversionValueFromDataResponse:(NSDictionary *) dataResponseDictionary{
+    
+    NSString *coarseConversionValue = dataResponseDictionary[BRANCH_RESPONSE_KEY_COARSE_KEY] ;
+ 
+    if (!coarseConversionValue) 
+        return @"low";
+        
+    return coarseConversionValue;
+  
+}
+
+- (BOOL) getLockedStatusFromDataResponse:(NSDictionary *) dataResponseDictionary {
+    
+    BOOL lockWin = NO;
+    if([dataResponseDictionary[BRANCH_RESPONSE_KEY_UPDATE_IS_LOCKED] isKindOfClass:NSNumber.class])
+        lockWin = ((NSNumber *)dataResponseDictionary[BRANCH_RESPONSE_KEY_UPDATE_IS_LOCKED]).boolValue;
+    return lockWin;
+}
+
+- (BOOL) getAscendingOnlyFromDataResponse:(NSDictionary *) dataResponseDictionary {
+
+    BOOL ascendingOnly = YES;
+    if([dataResponseDictionary[BRANCH_RESPONSE_KEY_ASCENDING_ONLY] isKindOfClass:NSNumber.class])
+        ascendingOnly = ((NSNumber *)dataResponseDictionary[BRANCH_RESPONSE_KEY_ASCENDING_ONLY]).boolValue;
+    return ascendingOnly;
+}
+
+- (BOOL) shouldCallPostbackForDataResponse:(NSDictionary *) dataResponseDictionary {
+    
+    BOOL shouldCallUpdatePostback = YES;
```

Comment:
Minor thing, but I feel safer defaulting to NO on something like this and verifying that we should make the call.